apps/tenant-management-webapp-e2e/src/integration/comment/comment.step.ts

```diff
@@ -505,3 +505,339 @@ Then(
+        commentObj.commentDeleteIcon(rowNumber).shadow().find('button').click({ force: true });
+        break;
+      default:
+        expect(iconName).to.be.oneOf(['edit', 'delete']);
+    }
+  });
+});
+
+//Find a comment with content
+//Input: comment content
+//Return: row number if a record is found; zero if record isn't found
+function findComment(comment) {
+  return new Cypress.Promise((resolve, reject) => {
+    try {
+      let rowNumber = 0;
+      commentObj
+        .commentsListContents()
+        .then((contents) => {
+          for (let i = 0; i < contents.length; i++) {
+            if (contents[i].innerHTML.includes(comment)) {
+              rowNumber = i + 1;
+              break;
+            }
+          }
+        })
+        .then(() => {
+          Cypress.log({
+            name: 'Row number for the found record: ',
+            message: String(rowNumber),
+          });
+          resolve(rowNumber);
+        });
+    } catch (error) {
+      reject(error);
+    }
+  });
+}
+
+Then('the user views Edit comment modal', function () {
+  commentObj.editCommentModalHeading().invoke('text').should('eq', 'Edit comment');
+});
+
+When('the user enters {string} in Edit comment modal', function (content) {
+  commentObj
+    .editCommentModalComment()
+    .shadow()
+    .find('.goa-textarea')
+    .clear({ force: true })
+    .type(content, { force: true, delay: 200 });
+});
+
+When('the user clicks Save button in Edit comment modal', function () {
+  commentObj.editCommentModalSaveBtn().shadow().find('button').click({ force: true });
+  cy.wait(2000);
+});
+
+Then('the user views {string} shows on top of {string}', function (content1, content2) {
+  findComment(content1).then((rowNumber1) => {
+    findComment(content2).then((rowNumber2) => {
+      expect(Number(rowNumber1)).to.lt(Number(rowNumber2));
+    });
+  });
+});
+
+Then('the user views Delete comment modal for {string}', function (comment) {
+  commentObj.deleteCommentModalHeading().invoke('text').should('eq', 'Delete comment');
+  commentObj.deleteCommentModalContent().invoke('text').should('contain', comment);
+});
+
+When('the user clicks Delete button in Delete comment modal', function () {
+  commentObj.deleteCommentModalDeleteBtn().shadow().find('button').click({ force: true });
+  cy.wait(2000);
+});
+
+Then('the user {string} the comment of {string}', function (viewOrNotView, comment) {
+  switch (viewOrNotView) {
+    case 'views':
+      commentObj.commentContent(comment).should('exist');
+      break;
+    case 'should not view':
+      commentObj.commentContent(comment).should('not.exist');
+      break;
+    default:
+      expect(viewOrNotView).to.be.oneOf(['views', 'should not view']);
+  }
+});
+
+Then('the user views Delete topic modal for {string}', function (topic) {
+  commentObj.deleteTopicModalHeading().invoke('text').should('eq', 'Delete topic');
+  commentObj.deleteTopicModalContentTopicName().invoke('text').should('contain', topic);
+});
```

Comment:
Delete modal also contains note: "*Please note that all associated comments with autotest-addDeleteTopic will be deleted as well." should that be added as part of verification?

Reply:
Thought about doing that, but didn't do that as it wasn't part of the manual test steps. Anyway, I will add that into the script.